src/datamodel/validator/UUri.cpp

```diff
@@ -10,3 +10,179 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/validator/UUri.h"
+
+namespace uprotocol::datamodel::validator::uri {
+
+using namespace uprotocol;
+
+std::string_view message(Reason reason) {
+	switch (reason) {
+		case Reason::EMPTY:
+			return "The URI authority_name is empty or whitespace.";
+		case Reason::RESERVED_VERSION:
+			return "The version is 0 (reserved).";
+		case Reason::RESERVED_RESOURCE:
+			return "The resource is 0 for non-RPC-response URIs.";
+		case Reason::DISALLOWED_WILDCARD:
+			return "The URI contains a wildcard in one or more fields that is "
+			       "not allowed for this form.";
+		case Reason::BAD_RESOURCE_ID:
+			return "The resource ID is not in the allowed range for this URI "
+			       "form.";
+		default:
+			return "Unknown reason.";
+	}
+}
+
+bool uses_wildcards(const v1::UUri& uuri) {
+	if (uuri.authority_name().find("*") != std::string::npos) {
+		return true;
+	}
+	if ((uuri.ue_id() & 0xFFFF) == 0xFFFF) {  // service ID
+		return true;
+	}
+	if ((uuri.ue_id() & 0xFFFF0000) == 0) {  // service instance ID
```

Comment:
@stevenhartley, correct me if am wrong, but Instance Id can be 0. As per someip (pr in review) spec if instance id is 0 it is treated as 1 and can not be treated as a wildcard entry.

Reply:
This is from a PR, not merged yet in mainline, and is part of someip spec
https://github.com/int0x27/up-spec/blob/someip-spec-update/up-l1/someip.adoc#21-uuris

Reply:
Make sure to verify ranges in the [uattributes spec](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uattributes.adoc). I believe 0 is excluded from most ranges in the URI, with the exception of resource where 0 is the destination for RPC responses

Reply:
It sounds like we need some clarity in the UURI specs. Let's get that first and we can revisit.